src/dtls.rs

```diff
@@ -148,6 +160,11 @@ pub fn parse_dtls_record_header(i: &[u8]) -> IResult<&[u8], DTLSRecordHeader> {
     Ok((i, record))
 }
 
+/// Treat the entire input as an opaque fragment.
+fn parse_dtls_fragment(i: &[u8]) -> IResult<&[u8], DTLSMessageHandshakeBody> {
+    Ok((&[], DTLSMessageHandshakeBody::Fragment(i)))
+}
+
```

Comment:
I understand we may not know the length of the fragment, but can't this consume more than needed? Can there be data after the fragment?

Reply:
Second thought: maybe this should be bound to at most `fragment_length` (from the caller)?

Reply:
If you check where this is called from, I think your concern is addressed.

```rs
 let (i, raw_msg) = take(fragment_length)(i)?;
```

`fragment_length` read from the header, controls the amount of data we consume from the input. It won't consume too much unless the DTLS header is malformed.

Reply:
Oh, I see, the function comment lead me to believe it was called with the entire output, while it is (and must be) called with raw fragment data.

Reply:
Can I improve the function name and/or maybe some code doc?

Reply:
This should be a comment in this function indicating the preconditions. However, I must admit some other (similar) functions do not all have comments